### DISCUSSION TITLE
Max key error

### DISCUSSION THREAD

**Discussion by theJohnCode** [spatie/laravel-permission#2378] [Q&A] [answered]
![error](https://user-images.githubusercontent.com/61396088/228657409-53dc3057-5697-43c5-8331-086ca04e8fce.png)
#1071 - Specified key was too long; max key length is 1000 bytes

How do I fix this. I am on laravel 9, Myql 10

**Comment by parallels999** [comment]
[Specified key was too long; max key length is 1000 bytes](https://www.google.com/search?q=mysql+key+was+too+long+max+key+length+is+1000+bytes)

**Comment by erikn69** [comment, marked as answer]
Duplicate of
https://github.com/spatie/laravel-permission/issues/909
https://github.com/spatie/laravel-permission/issues/2082
https://github.com/spatie/laravel-permission/issues/1689
https://github.com/spatie/laravel-permission/issues/1438
https://github.com/spatie/laravel-permission/issues/1108
https://github.com/spatie/laravel-permission/issues/218 and others
https://github.com/spatie/laravel-permission/blob/addbd4f9e2a0933526ec75cfd77a3ef5ef1aae36/database/migrations/create_permission_tables.php.stub#L31-L32
https://github.com/spatie/laravel-permission/blob/addbd4f9e2a0933526ec75cfd77a3ef5ef1aae36/database/migrations/create_permission_tables.php.stub#L44-L45
